src/Compilers/CSharp/Portable/CSharpParseOptions.cs

```diff
@@ -83,7 +88,7 @@ public override IEnumerable<string> PreprocessorSymbolNames
         }
 
         private CSharpParseOptions(CSharpParseOptions other) : this(
-            languageVersion: other.LanguageVersion,
+            languageVersion: other.SpecifiedLanguageVersion,
```

Comment:
Add one then, we already have and Internal constructor that doesn't do any validation on the arguments. It also follow a more Immutable approach.